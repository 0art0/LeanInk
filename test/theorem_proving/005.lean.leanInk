[{"contents":
  "theorem test (p q : Prop) (hp : p) (hq : q) : p ∧ q ∧ p := by\n  ",
  "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\np q : Prop hp : p hq : q ⊢ p ∧ q ∧ p",
    "_type": "goal"}],
  "contents": "apply And.intro hp",
  "_type": "sentence"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\np q : Prop hp : p hq : q ⊢ q ∧ p",
    "_type": "goal"}],
  "contents": ";",
  "_type": "sentence"},
 {"contents": " ", "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses": [],
    "conclusion": "\np q : Prop hp : p hq : q ⊢ q ∧ p",
    "_type": "goal"}],
  "contents": "exact And.intro hq hp",
  "_type": "sentence"},
 {"contents": "\n\n#check", "_type": "text"}]