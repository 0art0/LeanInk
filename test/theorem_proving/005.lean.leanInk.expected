[{"contents": "theorem test (p q : Prop) (hp : p) (hq : q) : p ‚àß q ‚àß p := ",
  "_type": "text"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "p ‚àß q ‚àß p",
    "_type": "goal"}],
  "contents": "by\n  ",
  "_type": "sentence"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "q ‚àß p",
    "_type": "goal"}],
  "contents": "apply And.intro hp",
  "_type": "sentence"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "q ‚àß p",
    "_type": "goal"}],
  "contents": ";",
  "_type": "sentence"},
 {"messages": [],
  "goals":
  [{"name": "",
    "hypotheses":
    [{"type": "Prop", "names": ["p", "q"], "body": "", "_type": "hypothesis"},
     {"type": "p", "names": ["hp"], "body": "", "_type": "hypothesis"},
     {"type": "q", "names": ["hq"], "body": "", "_type": "hypothesis"}],
    "conclusion": "p ‚àß q ‚àß p",
    "_type": "goal"}],
  "contents": " ",
  "_type": "sentence"},
 {"messages": [],
  "goals": [{"name": "", "hypotheses": [], "conclusion": "üêô", "_type": "goal"}],
  "contents": "exact And.intro hq hp",
  "_type": "sentence"},
 {"contents": "\n\n", "_type": "text"},
 {"messages": [{"contents": "And : Prop ‚Üí Prop ‚Üí Prop", "_type": "message"}],
  "goals": [],
  "contents": "#check",
  "_type": "sentence"},
 {"contents": " And", "_type": "text"}]